app/scripts/contentscript.js

```diff
@@ -11,32 +11,58 @@ const inpageContent = fs.readFileSync(path.join(__dirname, '..', '..', 'dist', '
 const inpageSuffix = '//# sourceURL=' + extension.extension.getURL('inpage.js') + '\n'
 const inpageBundle = inpageContent + inpageSuffix
 
+
+const inpageWeb3Content = fs.readFileSync(path.join(__dirname, '..', '..', 'dist', 'chrome', 'inpage-web3.js')).toString()
+const inpageWeb3Suffix = '//# sourceURL=' + extension.extension.getURL('inpage-web3.js') + '\n'
+const inpageWeb3Bundle = inpageWeb3Content + inpageWeb3Suffix
+
 // Eventually this streaming injection could be replaced with:
 // https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Language_Bindings/Components.utils.exportFunction
 //
 // But for now that is only Firefox
 // If we create a FireFox-only code path using that API,
 // MetaMask will be much faster loading and performant on Firefox.
 
-if (shouldInjectWeb3()) {
-  setupInjection()
-  setupStreams()
+if (shouldInjectProvider()) {
+  listenForProviderRequest()
+}
+
+/**
+ * Establishes listeners for provider injection requests from the browser context
+ * and for provider approvals and rejections from the background script context
+ */
+function listenForProviderRequest () {
+  window.addEventListener('message', (event) => {
+    if (event.source !== window) { return }
+    if (!event.data || !event.data.type || event.data.type !== 'ETHEREUM_PROVIDER_REQUEST') { return }
```

Comment:
To enable better support for multiple wallets metamask should only handle the event if the `id` is not set or `"METAMASK"` (see https://eips.ethereum.org/EIPS/eip-1102 `[1] REQUEST`)

Reply:
Great catch, thanks @rmeissner. Will update.